src/MyHealth/symptoms.ts

```diff
@@ -22,3 +45,81 @@ export const determineHealthAssessment = (
   }
   return HealthAssessment.NotAtRisk
 }
+
+type WithADate = { date: Posix }
+const compareDatesAscending = (
+  { date: dateLeft }: WithADate,
+  { date: dateRight }: WithADate,
+) => {
+  return dateLeft < dateRight ? -1 : 1
+}
+const compareDatesDescending = (
+  { date: dateLeft }: WithADate,
+  { date: dateRight }: WithADate,
+) => {
+  return dateLeft < dateRight ? 1 : -1
+}
+
+const groupLogEntriesDaily = (allEntries: SymptomLogEntry[]): LogDataByDay => {
+  const groupedEntries: LogDataByDay = {}
+
+  allEntries.forEach((entry) => {
+    const { date } = entry
+    const entryDateBeginningOfDay = beginningOfDay(date)
+
+    if (groupedEntries[entryDateBeginningOfDay]) {
+      const newLogEntries = groupedEntries[entryDateBeginningOfDay].logEntries
+      newLogEntries.push(entry)
+      newLogEntries.sort(compareDatesAscending)
+      groupedEntries[entryDateBeginningOfDay].logEntries = newLogEntries
+    } else {
+      groupedEntries[entryDateBeginningOfDay] = {
+        logEntries: [entry],
+        checkIn: null,
+      }
+    }
+  })
+
+  return groupedEntries
+}
+
+const joinLogDataAndDailyCheckIns = (
+  logData: LogDataByDay,
+  dailyCheckIns: DailyCheckIn[],
+): LogDataByDay => {
+  dailyCheckIns.forEach((checkIn) => {
+    const { date } = checkIn
+    const checkInBeginningOfDay = beginningOfDay(date)
+
+    if (logData[checkInBeginningOfDay]) {
+      logData[checkInBeginningOfDay].checkIn = checkIn
+    } else {
+      logData[checkInBeginningOfDay] = {
+        logEntries: [],
+        checkIn,
+      }
+    }
+  })
+  return logData
+}
+
+export const serializeDailyLogData = (
+  symptomLogEntries: SymptomLogEntry[],
+  dailyCheckIns: DailyCheckIn[],
+): DayLogData[] => {
```

Comment:
Having a mix of Log / Entry / Data is a bit confusing
What do you think about

```suggestion
export const combineEntryData = (
  symptomLogEntries: SymptomLogEntry[],
  checkInEntries: CheckInEntry[],
): DailyEntry[] => {
```

or even

```suggestion
export const combineSymptomAndCheckInLogs = (
  symptomLogs: SymptomLog[],
  checkInLogs: CheckInLog[],
): DailyLog[] => {
```

?

Reply:
💯 behind the renaming, gonna merge this and start making those changes